From d6db21c8722466b8b38cdaed8582329567efc031 Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Thu, 27 Mar 2025 15:21:12 +0000
Subject: [PATCH] wayland: Avoid creating the platform window in surfacehelper

All window effect code has code to dynamically handle the window
being created and torn down to update the wayland state. We
don't need or want to explictly create a window ahead of time.

This is proving problematic as there is a code-path that causes
us to re-enter QWindow::create unsafely.

CCBUG: 499383
---
 src/platforms/wayland/surfacehelper.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/platforms/wayland/surfacehelper.h b/src/platforms/wayland/surfacehelper.h
index 7964688c..6d25a193 100644
--- a/src/platforms/wayland/surfacehelper.h
+++ b/src/platforms/wayland/surfacehelper.h
@@ -23,6 +23,5 @@ inline wl_surface *surfaceForWindow(QWindow *window)
     if (!native) {
         return nullptr;
     }
-    window->create();
     return reinterpret_cast<wl_surface *>(native->nativeResourceForWindow(QByteArrayLiteral("surface"), window));
 }
-- 
GitLab

